src/twisted/internet/test/test_process.py

```diff
@@ -843,6 +843,26 @@ def start():
         reactor.callWhenRunning(start)
         self.runReactor(reactor)
 
+    def test_pauseAndResumeProducing(self):
+        """
+        Pause producing and then resume producing.
+        """
+
+        def pauseAndResume(reactor):
+            try:
+                protocol = ProcessProtocol()
+                transport = reactor.spawnProcess(
+                    protocol, pyExe, [pyExe, b"-c", b""], usePTY=self.usePTY
+                )
+                transport.pauseProducing()
+                transport.resumeProducing()
```

Comment:
we should test this streaming to an object with actual flow control

Reply:
here's a test using a MemoryProcessReactor that could be updated to use a real reactor

https://github.com/twisted/twisted/blob/c6c863bf8430f5f1e017af874ffdc89b2cc1b710/src/twisted/internet/test/test_endpoints.py#L822-L1140

Reply:
I'm not totally clear about this review. Is this a request to change the test in the PR?

Reply:
To add an additional test that uses a real reactor and process

Reply:
I think the test in the PR does use a real reactor and a real process already.  It uses the "reactor builder" testing framework which runs the test against *all* of the real reactors, even.

Reply:
ah ok, that makes sense